lnwallet/channel.go

```diff
@@ -3896,16 +4041,32 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 		lc.channelState.RemoteChanCfg.ChanReserve,
 	)
 
+	// Calculate the commitment fee to log the information if needed.
+	commitFee := feePerKw.FeeForWeight(commitWeight)
+	commitFeeMsat := lnwire.NewMSatFromSatoshis(commitFee)
+
 	switch {
+	// TODO(ziggie): Allow the peer dip us below the channel reserve when
```

Comment:
I think you are right we definitly need to update the spec here to make sure we force-close the channel if this happens instead of just sending out a warning, because the channel is done. But I am not sure whether we should accept the HTLC which implies sending the `update_fail_htlc `, because we don't want to advance the channelstate if it dips us below the reserve. Maybe we need to special case it when we introduce the exception for splicing will talk about it in the BOLT Spec Meeting.

Reply:
A second thought is, it's OK to send `warning` or `error` to stop the link - on our end, we'd make sure it won't happen, and if they make it happen, then it's somewhat reasonable to think the peer is violating the protocol so we can terminate - though in reality we know it's better to loosen it a bit.

On the other hand, I think we should definitely distinguish the case when the channel drops below fee buffer vs drops below channel reserve. We should never fail the link when it drops below fee buffer, since that's what the buffer is used for - to buffer the concurrent add updates that may happen to a given channel. So imo we should somehow cancel that htlc if it will make the channel drop below fee buffer.

Re `update_fail_htlc`, that's a very good point. I think if we are the sender of the htlc, we can easily cancel it back to the upper subsystems instead of sending an `error` to the peer. And if we are the receiver of the htlc, since we don't enforce the fee buffer check, we cannot do anything. But this sort makes me wondering we should also check the fee buffer for the remote balance, and sends back a `warning` message.

In addition, we should treat `error` and `warning` messages differently, need to check how we handle them more.

Reply:
> On the other hand, I think we should definitely distinguish the case when the channel drops below fee buffer vs drops below channel reserve. We should never fail the link when it drops below fee buffer, since that's what the buffer is used for - to buffer the concurrent add updates that may happen to a given channel. So imo we should somehow cancel that htlc if it will make the channel drop below fee buffer.

Agree currently we make sure that we only enforce the buffer when we are in control of also aborting the htlc process, so I think we already do it but its currently not set into stone meaning that the `validateAddHtlc` can be called with any buffer in the future so maybe we should introduce extra functions because of the safety maybe ?